src/components/molecules/ChatList/ChatList.tsx

```diff
@@ -84,22 +87,38 @@ const ChatList: React.FC<ChatListProps> = ({
   const hideDeleteModal = useCallback(() => setShowDeleteModal(false), []);
 
   const hasMessages = hasElements(messages);
+
   const messageSender = usersById?.[messageToDelete?.from ?? ""]?.partyName;
+
+  const renderMessages = useMemo(() => {
+    // Last (newest) message goes first
+    const sortedMessages = sortBy(messages, ["ts_utc"]).reverse();
+
+    return sortedMessages.map((message, index) => (
+      <ChatMessage
+        key={`${message.from}-${message.ts_utc.seconds}`}
```

Comment:
If someone sends 2 messages in less than a second, does that mean that there will be duplicated keys?

Reply:
I believe it will. Do we have milliseconds available on that object?